go/mysql/constants.go

```diff
@@ -677,3 +678,33 @@ func IsSchemaApplyError(err error) bool {
 	}
 	return false
 }
+
+type ReplicationState int
+
+const (
+	ReplicationStateRunning ReplicationState = iota
+	ReplicationStateStopped
+	ReplicationStateConnecting
+	ReplicationStateUnknown
+)
+
+// ReplicationStatusToState converts a value you have for the IO thread(s) or SQL
+// thread(s) or Group Replication applier thread(s) from MySQL or intermediate
+// layers to a mysql.ReplicationState.
+// on,yes,true == ReplicationStateRunning
+// off,no,false == ReplicationStateStopped
+// connecting == ReplicationStateConnecting
+// anything else == ReplicationStateUnknown
+func ReplicationStatusToState(s interface{}) ReplicationState {
+	status := strings.ToLower(fmt.Sprintf("%v", s))
+	// Group Replication uses ON instead of Yes
+	if status == "yes" || status == "on" || status == "true" {
+		return ReplicationStateRunning
+	} else if status == "no" || status == "off" || status == "false" {
+		return ReplicationStateStopped
+	} else if status == "connecting" {
+		return ReplicationStateConnecting
+	} else {
+		return ReplicationStateUnknown
+	}
```

Comment:
purely a styling matter, but I think `switch` is cleaner here:
```go
	switch strings.ToLower(s) {
	case "yes", "on", "true":
		return ReplicationStateRunning
	case "no", "off", "false":
		return ReplicationStateStopped
	case "connecting":
		return ReplicationStateConnecting
	default:
		return ReplicationStateUnknown
	}
```
feel free to ignore any styling comment of mine.

Reply:
Sure, I don't have a strong preference either way. I started on this patch and then switched and don't remember why. 🙂 Was making too many changes while tired, I think. I'll move to this. Thanks!